drivers/modem/quectel-bg9x.c

```diff
@@ -735,9 +736,18 @@ static int offload_connect(void *obj, const struct sockaddr *addr,
 
 	k_sem_reset(&mdata.sem_sock_conn);
 
+	ret = modem_context_sprint_ip_addr(addr, ip_str, sizeof(ip_str));
+	if (ret != 0) {
+		LOG_ERR("Error formatting IP string %d", ret);
+		LOG_ERR("Closing the socket!!!");
+		socket_close(sock);
+		errno = -ret;
+		return -1;
+	}
+
 	/* Formulate the complete string. */
 	snprintk(buf, sizeof(buf), "AT+QIOPEN=%d,%d,\"%s\",\"%s\",%d,0,0", 1, sock->sock_fd, protocol,
-		 modem_context_sprint_ip_addr(addr), dst_port);
+			ip_str, dst_port);
```

Comment:
Nit (styling issue): the beginning of this line is no longer aligned with the `(` from the previous line.
```suggestion
		 ip_str, dst_port);
```

Reply:
Fixed